### PR TITLE
tapir API docs wiki generator

### PR DESCRIPTION
**Pull request by harryurban**
# What
- [Dockerfile] add binary for protoc for generating docs from proto files
- [Makefile] add target to generate docs for all of /stroeer in one `api-spec.md`
- [CI] add workflow to add a link into the index file `api-spec.md` to the specs for the release

# Why
- Will produce this: https://github.com/stroeer/tapir/wiki/api-spec
- #84

# Comment
- generated docu relies on proper comments in files
  - see an example here:  https://github.com/stroeer/tapir/pull/83
- syntax for comments:
  - https://github.com/pseudomuto/protoc-gen-doc#writing-documentation

# Beware
- checks fail because docker image with the new binary was not pushed yet

### CODE REVIEW

File: .github/workflows/api-docs.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          cd tapir.wiki
+          [ -d $SPEC_FOLDER ] || mkdir -p $SPEC_FOLDER
+          yes | cp ../api-docs/api-spec* $SPEC_FOLDER
+          cd $SPEC_FOLDER
+          echo "\n\n## $RELEASE_TAG \n" >> ../../api-spec.md
+          for n in api-spec-stroeer*; do filepath=${n/.md/}; label=${filepath/api-spec-/}; echo "• [$label]($SPEC_FOLDER/$filepath)" >> ../../api-spec.md; done
+          cd ../..
```
Comment (harryurban) [inline .github/workflows/api-docs.yml:30]: I am sure there is much easier and shorter code for this part. so feedback is very welcome.

File: Dockerfile [hunk 2]
```diff
@@ -42,7 +48,8 @@ COPY --from=gopher /go/bin/protoc-gen-go-grpc /usr/bin/
 COPY --from=java /installer/protoc-gen-grpc-java /usr/bin/
 COPY --from=node /node_modules /node_modules
 COPY --from=protoc-bin /installer/protoc /usr/bin/protoc
-
+COPY --from=protoc-bin /installer/protoc-gen-doc /usr/bin/protoc-gen-doc
+RUN chmod +x /usr/bin/protoc-gen-grpc-java
 RUN chmod +x /usr/bin/protoc-gen-grpc-java
```
Comment (bobaaaaa) [inline Dockerfile:53]: duplicate?

Comment (harryurban) [reply Dockerfile:53]: Correct. Should be other filename but since it works like that I will remove it

File: Makefile [hunk 3]
```diff
@@ -51,29 +53,40 @@ endif
...
 	[ -d $@ ] || mkdir -p $@
 
-stroeer/%: $(OUTPUT)
-	@echo "+ $@"
-	$(PROTOC) $(FLAGS) $(shell find $@ -iname "*.proto" -exec echo -n '"{}" ' \; | tr '\n' ' ')
+
+stroeer/%: $(OUTPUT) $(DOCS_DIR) clean-docs
+	@echo "+ compile: $@"
+	$(eval doc_name := api-spec-$(subst /,.,$@))
+	$(PROTOC) $(FLAGS) --doc_opt=markdown,$(doc_name).md $(shell find $@ -iname "*.proto" -exec echo -n '"{}" ' \; | tr '\n' ' ')
```
Comment (harryurban) [inline Makefile:67]: Will change to an own target

**Comment by bobaaaaa** [comment]
I already talked to Harry. I personally would prefer a github pages deployment (not into the wiki) because:

- its versioned as a whole
- its unchangeable

File: Makefile (hunk 3 above)
Comment (harryurban) [reply Makefile]: done

**Comment by harryurban** [comment]
draft implementation pr is closed for now:
- `protoc-gen-doc` library is not up to date with proto3 (doesn't know `oneof` for example)
- usecases for a markdown docu are limited

maybe this will PR be replaced with a proper and robust solution as a hobby project later on

**Pull request closed without merging**